cpp/src/io/json/write_json.cu

```diff
@@ -268,40 +270,53 @@ std::unique_ptr<column> struct_to_strings(table_view const& strings_columns,
   auto const num_columns = strings_columns.num_columns();
   CUDF_EXPECTS(num_columns == column_names.size(),
                "Number of column names should be equal to number of columns in the table");
-  auto const strings_count = strings_columns.num_rows();
-  if (strings_count == 0)  // empty begets empty
+  if (num_rows == 0)  // empty begets empty
     return make_empty_column(type_id::STRING);
   // check all columns are of type string
   CUDF_EXPECTS(std::all_of(strings_columns.begin(),
                            strings_columns.end(),
                            [](auto const& c) { return c.type().id() == type_id::STRING; }),
                "All columns must be of type string");
   auto constexpr strviews_per_column = 3;  // (for each "column_name:", "value",  "separator")
-  auto const num_strviews_per_row    = strings_columns.num_columns() * strviews_per_column + 1;
+  auto const num_strviews_per_row    = 1 + strings_columns.num_columns() * strviews_per_column +
+                                    (strings_columns.num_columns() == 0 ? 1 : 0);
```

Comment:
I can't tell what the first 1 is, and what the 1 added for `strings_columns.num_columns() == 0` is

Reply:
How about:
`strings_columns.num_columns() == 0 ? 2 : 1 + strings_columns.num_columns() * strviews_per_column;`?
2 for empty `{}`

Which is essentially `1 + strings_columns.num_columns() * strviews_per_column - 1 + 1`;
for ` {col1: value, col2: value, col3: value}` = 1 + 3 + 3 + 3-1 + 1
= 1`{` + 3 `col1: value ,` + 3 `col2: value` + 2 `col3: value` + 1`}`